themes/default/content/blog/pulumi-release-notes-m58/index.md

```diff
@@ -0,0 +1,337 @@
+
+- Refined, shorter resource names that are easier to type and easier to find
+- Enum support for all properties exposed as enums in the underlying Google Discovery Documents
+
+[See the full list of updates](https://github.com/pulumi/pulumi-google-native/blob/master/CHANGELOG.md#040-2021-06-16)
+
+## Pulumi CLI and core technologies
+
+In this milestone, we shipped Pulumi versions [3.5.1](https://github.com/pulumi/pulumi/releases/tag/v3.5.1) and [3.6.0](https://github.com/pulumi/pulumi/releases/tag/v3.6.0). The full list of changes in each version is available in the linked changelog; read on to learn about some of the biggest changes.
+
+### New `replaceOnChanges` and `deleteBeforeReplace` resource options
+
+There are times when it is important to force a replacement of a resource even if changes to the resource do not indicate they require replacement. One example is a Kubernetes `CustomResourceDefinition`, none of whose properties force replacement, even though for many CRDs, they can't actually be updated in place (because of forward-incompatibilites). While it was previously possible to work around the lack of an option to force a replace by changing the physical name of the resource, there are cases where it is important to keep the physical name the same.
+
+Now, you can use the new `replaceOnChange` resource option to force a resource to be replaced when the specified properties are changed. You can also use the `deleteBeforeReplace` resource option to force a resource to be deleted before it's replaced. For example, you can specify that an AWS S3 Bucket resource be replaced every time its index document changes:
+
+```typescript
+const bucket = new aws.s3.Bucket("my-bucket", {
+    website: {
+        indexDocument: "index2.html",
+    },
+}, { replaceOnChanges: ["website.indexDocument"], deleteBeforeReplace: true });
+```
+
+Learn more at [pulumi/pulumi#7226](https://github.com/pulumi/pulumi/pull/7226)
+
+### `pulumi watch` now supports a `--path` option
+
+Previously, the `pulumi watch` command always watched the entire Pulumi project directory (recursively). This could cause duplicate updates for developers working in monorepos or developers with application code and infrastructure code in the same directory. Now, you can add a `--path` option to the `pulumi watch` command and specify the exact paths that need to be watched.
+
+```shell
+pulumi watch --path path/to/pulumi/code/
+```
+
+Learn more at [pulumi/pulumi#6620](https://github.com/pulumi/pulumi/issues/6620)
+
+### Automation API: support for custom loggers in .NET
+
+Many mature projects already have a preferred tool for logging events and exceptions. Previously, Automation API only supported a built-in logging tool, and it wasn't possible to use a project's preferred tool. Now, if you're using a .NET language like C#, you can supply your own logging tool, like Serilog.
```

Comment:
I'm not sure this is an accurate description, but I'm also not entirely sure I follow what the improvement was here.  cc @t0yv0 for input to clarify what this should say (or whether this is small enough we should leave it out).

Reply:
It's a community PR permitting using and customizing Serilog which enables a nice IDE integration for tests; I think it's small enough to leave out of notes here. If we want to include I can dig up the relevant details tomorrow.

Reply:
@t0yv0 thanks, I'll remove this item